source/js/plugins/pixel-wrapper.js

```diff
@@ -150,30 +208,31 @@ export class PixelWrapper
             layerCanvas.setAttribute("style", "position: absolute; top: 0; left: 0;");
             layerCanvas.width = width;
             layerCanvas.height = height;
-            layer.drawLayerInPageCoords(maxZoom, layerCanvas, pageIndex); 
+            layer.drawLayerInPageCoords(maxZoom, layerCanvas, this.pageIndex); 
 
-            this.subtractLayerFromBackground(backgroundLayer, layerCanvas, pageIndex, width, height);
+            this.subtractLayerFromBackground(backgroundLayer, layerCanvas, width, height);
         });
     }
 
-    subtractLayerFromBackground (backgroundLayer, layerCanvas, pageIndex, width, height) 
+    subtractLayerFromBackground (backgroundLayer, layerCanvas, width, height) 
     {
         var chunkSize = width,
             chunkNum = 0,
             row = 0,
             col = 0,
             pixelCtx = layerCanvas.getContext('2d');
-        let doChunk = () => { // Use this method instead of nested for so UI isn't blocked
+
+        let doChunk = () => { 
             var cnt = chunkSize;
             chunkNum++;
             while (cnt--) { 
                 if (row >= height)
                     break;
                 if (col < width) {
-                    let data = pixelCtx.getImageData(col, row, 1, 1).data,
-                        colour = new Colour(data[0], data[1], data[2], data[3]);
-                    if (colour.alpha !== 0) { 
-                        let currentPixel = new Rectangle(new Point(col, row, pageIndex), 1, 1, "subtract");
+                    let data = pixelCtx.getImageData(col, row, 1, 1).data;
+                    // data[3] is alpha
```

Comment:
Does the ```data``` array follow RGB+alpha? Please remind that in the code.

Reply:
Done!